
import "Place.proto";
import "CityOverview.proto";
import "TravelTips.proto";
import "App.proto";
import "TouristRoute.proto";

package damuzhi;

option java_package = "com.damuzhi.travel";
option java_outer_classname = "PackageProtos";

enum LanguageType{
  ZH_HANS = 1;          // 简体（Simplified）
  ZH_HANT = 2;          // 繁体（Tranditional）
  EN = 3;               // 英文（未来扩展）
}

message Package {
  required string version = 1;                                     // 数据包格式版本，如1.0
  required int32  cityId = 2;                                      // 城市ID
  required string cityName = 3;                                    // 城市名称
  required LanguageType language = 4 [default=ZH_HANS];            // 语言类型
}

// 用于会员资料
message UserInfo {
  required string userId = 1;		// 用户id，与设备id绑定
  required int32  loginType = 2;    // 当前登录类型，1：手机；2：邮箱
  required string telephone = 3;	// 电话
  required string email = 4; 		// 邮箱
  required string password = 5;		// 密码

  optional string nickName = 6;		// 昵称
  optional int32  gender = 7; 		// 性别
  optional string fullName = 8;		// 姓名
}

// 用于获取线路反馈列表
message RouteFeekbackList
{
  repeated RouteFeekback routeFeekbacks = 1;
}

// 用于路线反馈
message RouteFeekback {
  required int32  routeId = 1;		// 线路编号
  required string userId = 2;		// 用户名，即会员的登陆时的用户名。只有会员才能评价路线。
  required string nickName = 3;		// 用户昵称
  required int32 date = 4;          // 评价时间，从1970年1月1日0时0分0秒算起，经过的秒数。
  required int32 rank = 5;          // 评级
  optional string content = 6;      // 反馈内容
}

message TravelResponse{
  required int32 resultCode = 1 [default=0];
  optional string resultInfo = 2;
  optional Place place = 3;
  optional CommonOverview overview = 4;
  optional CommonTravelTip travelTip = 5;
  optional HelpInfo helpInfo = 6;
  optional PlaceList placeList = 7;
  optional CityList cityList = 8;
  optional App appInfo = 9;   
  optional CommonTravelTipList travelTipList = 10;
  
  optional UserInfo userInfo = 15;                      // 用户资料
  
  optional TouristRoute route = 20;                     // 线路详情
  optional TouristRouteList routeList = 21;             // 线路列表
  
  optional RouteFeekbackList routeFeekbackList= 30;     // 线路反馈
  
  optional OrderList orderList = 35;                    // 订单列表
}
